### PR TITLE
build: update react to version 17

### PR DESCRIPTION
**Pull request by gdostie**
### Proposed Changes

Update react to version 17

### Potential Breaking Changes

None

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

**Comment by aiheon** [comment]
I would commit if I could but you'll need to put pnpm@6 in the jenkinfiles 😓

**Comment by aiheon** [comment]
I've rolled back to V6 locally so I was able to commit my changes in another branch, i'll merge soon you'll just have to rebase 👍

**Comment by aiheon** [comment]
it's ready to rebase! 🎉

File: packages/react-icons/package.json [hunk 1]
```diff
@@ -46,7 +52,9 @@
         "lodash.upperfirst": "4.3.1",
         "npm-run-all": "4.1.5",
         "tslib": "2.3.1",
-        "typescript": "4.6.4"
+        "typescript": "4.6.4",
+        "react": "17.0.2",
+        "react-dom": "17.0.2"
```
Comment (GermainBergeron) [inline packages/react-icons/package.json:57]: The order is wrong, react stuff should be before tslib 🙈

File: packages/react-icons/package.json [hunk 2]
```diff
@@ -33,7 +33,13 @@
     "files": [
         "dist"
     ],
+    "peerDependencies": {
+        "react": ">= 17.0.2",
+        "react-dom": ">= 17.0.2"
+    },
     "devDependencies": {
+        "@types/react": "17.0.44",
+        "@types/react-dom": "17.0.15",
```
Comment (GermainBergeron) [inline packages/react-icons/package.json:42]: We already have @types/react on line 45

File: package.json [hunk 3]
```diff
@@ -63,5 +63,13 @@
...
             "@commitlint/config-conventional"
         ]
+    },
+    "pnpm": {
+        "overrides": {
+            "react": "17.0.2",
+            "react-dom": "17.0.2",
+            "@types/react": "17.0.44",
+            "@types/react-dom": "17.0.15"
+        }
```
Comment (GermainBergeron) [inline package.json:73]: I guess it doesn't work without those?

File: packages/react/package.json [hunk 4]
```diff
@@ -131,9 +130,9 @@
         "lorem-ipsum": "2.0.4",
         "minimist": "1.2.6",
         "postcss": "8.4.13",
-        "react": "16.14.0",
         "react-dnd-test-backend": "15.1.2",
-        "react-dom": "16.14.0",
+        "react": "17.0.2",
+        "react-dom": "17.0.2",
```
Comment (GermainBergeron) [inline packages/react/package.json:135]: The order of dependencies is wrong here too

File: package.json (hunk 3 above)
Comment (gdostie) [reply package.json:73]: Nope, see [build #3](https://jsadminbuilds.dev.cloud.coveo.com/job/plasma/job/PR-2718/3/)

**Review by aiheon — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (gdostie) [reply package.json]: After the update to pnpm 7, it works without the overrides 👍

Comment (aiheon) [reply package.json]: 🎉

File: packages/react/package.json [hunk 5]
```diff
@@ -84,70 +84,69 @@
...
         "@types/react-color": "3.0.6",
-        "@types/react-dom": "16.9.15",
+        "@types/react-dom": "17.0.15",
         "@types/react-infinite-scroll-component": "4.2.5",
         "@types/react-modal": "3.13.1",
         "@types/react-redux": "7.1.22",
         "@types/react-tether": "0.5.5",
         "@types/react-textarea-autosize": "4.3.6",
         "@types/react-transition-group": "2.9.2",
+        "@types/react": "17.0.44",
```
Comment (GermainBergeron) [inline packages/react/package.json:103]: The order of dependencies is still wrong 🙊 

"@types/react" comes before "@types/react-dom"

same for "@types/underscore" and "@types/underscore.string"

I guess you can just run `pnpm --recursive update @types/react@17.0.44` and see if pnpm fixes everything

**Review by GermainBergeron — APPROVED** [review]

File: packages/react/package.json (hunk 5 above)
Comment (gdostie) [reply packages/react/package.json:103]: I sorted the lines with vscode and that's the order it chose 🤷

**Pull request merged**